atoma-state/src/state_manager.rs

```diff
@@ -230,13 +232,39 @@ impl AtomaState {
         fields(task_small_id = %task_small_id)
     )]
     pub async fn get_task_by_small_id(&self, task_small_id: i64) -> Result<Task> {
-        let task = sqlx::query("SELECT * FROM tasks WHERE task_small_id = ?")
+        let task = sqlx::query("SELECT * FROM tasks WHERE task_small_id = $1")
             .bind(task_small_id)
             .fetch_one(&self.db)
             .await?;
         Ok(Task::from_row(&task)?)
     }
 
+    pub async fn get_stacks_for_model(&self, model: &str, free_units: i32) -> Result<Vec<Stack>> {
+        let tasks = sqlx::query(
+            "SELECT stacks.* FROM stacks
+                INNER JOIN tasks ON tasks.task_small_id = stacks.task_small_id 
+                WHERE tasks.model_name = $1 AND stacks.num_compute_units - stacks.already_computed_units >= $2")
```

Comment:
I believe the `model_name` is not sufficient to retrieve the right stacks, as we also care about the `security level`, and other parameters. But for now, this looks okay. I suggest adding an issue, for better filtering, though